tls/s2n_client_hello.c

```diff
@@ -116,9 +116,15 @@ int s2n_client_hello_cb_done(struct s2n_connection *conn)
     POSIX_ENSURE(conn->config->client_hello_cb_mode ==
         S2N_CLIENT_HELLO_CB_NONBLOCKING, S2N_ERR_INVALID_STATE);
     POSIX_ENSURE(conn->client_hello.callback_invoked == 1, S2N_ERR_ASYNC_NOT_PERFORMED);
+    POSIX_ENSURE(conn->client_hello.parsed == 1, S2N_ERR_INVALID_STATE);
 
     conn->client_hello.callback_async_blocked = 0;
     conn->client_hello.callback_async_done = 1;
+    /*
+     * Disable polling once done. This prevents the callback from being called
+     * again if a hello retry is requested.
+     */
+    conn->config->callback_enable_poll = 0;
```

Comment:
The connection shouldn't be modifying the config. The way you've got it now, you should just set this value once when you build the config.

Reply:
In other words, just remove this section.

Reply:
@lrstewart mentioned that we didnt want to invoke the callback a second time when handling the hello retry. This change in this commit achieves this. https://github.com/aws/s2n-tls/pull/3230/commits/1bb2f7033e809d0704cb2875d4224b68b72723aa 

If we want to say that polling will behave differently then we can also go that route.